src/audio/src/src.c

```diff
@@ -592,6 +591,27 @@ static void src_set_sink_params(struct comp_dev *dev, struct comp_buffer __spars
 #error "No or invalid IPC MAJOR version selected."
 #endif /* CONFIG_IPC_MAJOR_4 */
 
+static inline void src_update_buffer_position(struct input_stream_buffer *input_buffers,
+					      struct output_stream_buffer *output_buffers,
+					      int *n_read, int *n_written)
+{
+	struct audio_stream __sparse_cache *source = input_buffers->data;
+	struct audio_stream __sparse_cache *sink = output_buffers->data;
+
+	input_buffers->consumed += audio_stream_frame_bytes(source) * (*n_read);
+	output_buffers->size += audio_stream_frame_bytes(sink) * (*n_written);
+}
+
+static void src_set_alignment(struct audio_stream __sparse_cache *source,
+			      struct audio_stream __sparse_cache *sink)
+{
+	const uint32_t byte_align = 1;
+	const uint32_t frame_align_req = 1;
+
+	audio_stream_init_alignment_constants(byte_align, frame_align_req, source);
+	audio_stream_init_alignment_constants(byte_align, frame_align_req, sink);
```

Comment:
Does this work with XCC builds? IPC3 isn't dead just yet. Do those implementations not have stricter alignment requirements?

Reply:
Yep, I'm testing with IPC3 too and will do for SRC after I can get back to this topic. I'm still using IPC3 builds for tests since we don't have everything yet converted to module adapter and IPC4. I prefer to develop in smaller steps like legacy -> module_adapter -> IPC3 to IPC4. This is my personal work preference.

Reply:
I have tested this patch on i.MX8QM (platform "imx8") and it seems to be fine on GCC (XTOS and Zephyr) and XCC (Zephyr, support not yet upstream).

Reply:
> I have tested this patch on i.MX8QM (platform "imx8") and it seems to be fine on GCC (XTOS and Zephyr) and XCC (Zephyr, support not yet upstream).

Great! I just fixed some IPC4 issues and will test a bit more before update. Did you try also capture direction?

Reply:
I don't think we _have_ any SRC + capture topologies on i.MX so nothing to test. So only playback tested.

Reply:
I got a capture direction SRC fail in testbench that is IPC3 but I haven't replicated it in a real device, so I'm not sure it's a real issue.

Reply:
Yeah, thankfully @iuliana-prodan really was quick to point out that we do in fact have the capture direction as well. Just tested it, seems fine on my end. Maybe it takes multiple iterations to reproduce? (I only tried once on two rates where SRC has to do work)

Reply:
That's very good news! The pipelines are not fully accurate in testbench with file instead of host/dai. In my case fail happened immediately in prepare() where one of the requested rates was zero.